packages/internal/whenable.js

```diff
@@ -13,7 +15,13 @@ const watchPromise = vatData && vatData.watchPromise;
  */
 const rejectionMeansRetry = reason => isUpgradeDisconnection(reason);
 
-export const powers = harden({
+export const defaultPowers = harden({
   rejectionMeansRetry,
   watchPromise,
 });
+
+export const prepareWhenableModule = (zone, powers = {}) =>
+  rawPrepareWhenableModule(zone, { ...defaultPowers, ...powers });
+
+export const { E, watch, when, makeWhenableKit, makeWhenablePromiseKit } =
+  prepareWhenableModule(makeHeapZone());
```

Comment:
These exports implicitly use the heap zone, which usually makes sense. When there are a similar set of exports made using a durable zone co-existing within the same system, I'm curious how they interact. This is certainly an issue and possible source of awkwardness and surprise for other zone coexistence questions, but at least I understand the answers there: durable cannot point at non-durable. Here I am having a harder time thinking it through.

The use of the disconnect-sensitive `rejectionMeansRetry` and the kernel-provided `watchPromise` if available makes this seem more oriented towards the durable zone. At least the retry issue is consistent with an ephemeral client wanting to reconnect to durable whenable from an upgraded server.

Altogether, I think I'm confused about what level of abstraction this module is supposed to be at, and why it is in @agoric/internal rather than @agoric/whenable

Reply:
In practice, I removed the return of `makeWhenablePromiseKit`, but not of `watch` and `when`, at least not until @iomekam's PR have migrated to the new `E.when` and `E.watch` names.

Reply:
Makes sense. But perhaps vat-data would then be an even better home?

In any case resolving. But please consider vat-data.